CRM/Xcm/MatchingEngine.php

```diff
@@ -506,20 +506,21 @@ protected function addActivityToContact($contact_id, $activity_type_id, $subject
    */
   protected function addDetailToContact($contact_id, $entity, &$data, $as_primary = FALSE, &$data_update = NULL) {
     if (!empty($data[$entity])) {
-      // sort out location type
-      if (empty($data['location_type_id'])) {
-        $location_type_id = $this->config->defaultLocationType();
-      } else {
-        $location_type_id = $data['location_type_id'];
-      }
-
-      // get attribute
-      $attribute = strtolower($entity); // for email and phone that works
-      $sorting = 'is_primary desc';
+      // Params for a possible "create" API call.
+      $create_detail_call['contact_id'] = $contact_id;
+      // sort out location/website type
       if (strtolower($entity) == 'website') {
-        $attribute = 'url';
+        $create_detail_call['website_type_id'] = $data['website_type_id'] ?? $this->config->defaultWebsiteType();
```

Comment:
Does this need an Updater for explicitly setting the default website type to what used to be the de-facto default value until now?